src/executables/util_sa.c

```diff
@@ -2911,8 +2913,19 @@ synccoll_check (const char *db_name, int *db_obs_coll_cnt, int *new_sys_coll_cnt
 		  fprintf (stdout, "%s\n", db_get_string (&class_name));
 
 		  /* output query to fix schema */
-		  snprintf (query, sizeof (query) - 1, "ALTER TABLE [%s] " "COLLATE utf8_bin;",
-			    db_get_string (&class_name));
+		  strcpy (split_owner_name, db_get_string (&class_name));
+		  split_name = strchr (split_owner_name, '.');
+		  if (split_name != NULL)
+		    {
+		      *split_name++ = '\0';
+		      snprintf (query, sizeof (query) - 1, "ALTER TABLE [%s].[%s] COLLATE utf8_bin;", split_owner_name,
+				split_name);
+		    }
+		  else
+		    {
+		      snprintf (query, sizeof (query) - 1, "ALTER TABLE [%s] COLLATE utf8_bin;",
+				db_get_string (&class_name));
+		    }
```

Comment:
The query buffer obtained by performing snprintf (query,...) is being output to fprintf (f_stmt, "%s\n", query); just below.

How about using the fprintf() function instead of snprintf()?

Reply:
I fixed it (d17f0ff)